user/src/sagas/authSagas.js

```diff
@@ -1,5 +1,82 @@
-import {  all } from "redux-saga/effects";
+import {
+    takeEvery,
+    all,
+    call,
+    select,
+    put,
+    takeLatest,
+} from "redux-saga/effects";
+import { LOGIN_REQUEST, KEEP_LOGIN, REGISTER_REQUEST, REGISTER_SUCCESS } from '../actions/constant';
+import { loginSuccess, loginFailure, registerFailure, registerSuccess } from "../actions/auth.action";
+import axiosInstance from "../helper/axios";
+
+const getUser = (state) => state.auth;
+
+const logIn = async (user) => {
+    const response = await axiosInstance.post("/user/login", {
+        ...user,
+    });
+    return { token: response.data.token, user: response.data.user };
```

Comment:
```suggestion
    const { data: { token, user } } = await axiosInstance.post("/user/login", {
        ...user,
    });
    return { token, user };
```
1 cách viết khác đối với destructuring